bin/s2nc.c

```diff
@@ -543,8 +543,16 @@ int main(int argc, char *const *argv)
             echo(conn, sockfd);
         }
 
+        /* The following call can block on receiving a close_notify if we initiate the shutdown or if the */
+        /* peer fails to send a close_notify. */
+        /* TODO: However, we should expect to receive a close_notify from the peer and shutdown gracefully. */
+        /* Please see tracking issue for more detail: https://github.com/aws/s2n-tls/issues/2692 */
         s2n_blocked_status blocked;
-        s2n_shutdown(conn, &blocked);
+        int shutdown_rc = s2n_shutdown(conn, &blocked);
```

Comment:
Commented on the issue, fix would be something like

```
uint8_t shutdown_complete = 0;

while (shutdown_complete == 0) {
    if (poll(sockfd, 1, 500) > 0) {
        if (s2n_shutdown(conn, &blocked) == S2N_SUCCESS) {
            shutdown_complete = 1;
    } else {
        perror("poll");
    }
}
```

This will block until data is available on `sockfd`. Then `s2n_shutdown` will find the close_notify.